### DISCUSSION TITLE
package mismatch

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#296] [Q&A]
a user asked:
> when installing the ChildProject Python package on the virtual environment of Python, it seems Python version 3.8 does not match some modules in this package such as Numpy. I downgraded it to version 3.7 but I have other problems with the "wheel and PEP 517" for building dependencies of scs . Do you have any recommendations for Python versions to install this package?

I think this is once more the same issue we were discussing elsewhere, but a search for "wheel" didn't reveal anything in discussions -- sorry if my search was not good!

**Comment by lucasgautheron** [comment]
Not sure what the user means by "Python version 3.8 does not match some modules in this package such as Numpy".
The package is automatically tested on four python versions (3.6 to 3.9), and the requirements are met for each of theses versions.
Whenever some package version incompatibility occurs for a reason or another, the tests fail and I usually solve this by adding constraints to the requirements.

It's always difficult to solve these problems without more information about the environment of every user...

**Reply by lucasgautheron** [reply]
Regarding the scs issue: it can be solved by installing ChildProject>=0.0.2 e.g. by doing `pip3 install git+https://github.com/LAAC-LSCP/ChildProject.git`

scs is a requirement of cvxpy I think which itsel is a requirement of pygamma-agreement, which used to be a requirement of ChildProject but is no longer a requirement by default precisely due to the frequent installation issues it raised :)
